src/kernels/raw/launcher/kernelEnvVarsService.node.ts

```diff
@@ -96,52 +95,68 @@ export class KernelEnvironmentVariablesService {
-        // Reinitialize the PATH variables.
-        // The values in `PATH` found in the interpreter trumps everything else.
-        // If we have more PATHS, they need to be appended to this PATH.
-        // Similarly for `PTYHONPATH`
-        // Additionally the 'PATH' variable may have different case in each, so account for that.
-        let otherEnvPathKey = Object.keys(interpreterEnv).find((k) => k.toLowerCase() == 'path');
-        const processPathKey = Object.keys(mergedVars).find((k) => k.toLowerCase() == 'path') || otherEnvPathKey;
-        if (otherEnvPathKey && processPathKey) {
-            mergedVars[processPathKey] = interpreterEnv[otherEnvPathKey];
-        }
-        if (interpreterEnv['PYTHONPATH']) {
-            mergedVars['PYTHONPATH'] = interpreterEnv['PYTHONPATH'];
-        }
-        otherEnvPathKey = Object.keys(customEnvVars).find((k) => k.toLowerCase() == 'path');
-        if (otherEnvPathKey && customEnvVars[otherEnvPathKey]) {
-            this.envVarsService.appendPath(mergedVars, customEnvVars[otherEnvPathKey]!);
-        }
-        otherEnvPathKey = Object.keys(kernelEnv).find((k) => k.toLowerCase() == 'path');
-        if (otherEnvPathKey && kernelEnv[otherEnvPathKey]) {
-            this.envVarsService.appendPath(mergedVars, kernelEnv[otherEnvPathKey]!);
-        }
-        if (customEnvVars.PYTHONPATH) {
-            this.envVarsService.appendPythonPath(mergedVars, customEnvVars.PYTHONPATH);
-        }
-        if (kernelEnv.PYTHONPATH) {
-            this.envVarsService.appendPythonPath(mergedVars, kernelEnv.PYTHONPATH);
-        }
-        // Ensure the python env folder is always at the top of the PATH, this way all executables from that env are used.
-        // This way shell commands such as `!pip`, `!python` end up pointing to the right executables.
-        // Also applies to `!java` where java could be an executable in the conda bin directory.
-        if (interpreter) {
-            this.envVarsService.prependPath(mergedVars, path.dirname(interpreter.uri.fsPath));
-        }
 
-        // If user asks us to, set PYTHONNOUSERSITE
-        // For more details see here https://github.com/microsoft/vscode-jupyter/issues/8553#issuecomment-997144591
-        // https://docs.python.org/3/library/site.html#site.ENABLE_USER_SITE
-        if (this.configService.getSettings(undefined).excludeUserSitePackages) {
-            traceInfo(`Adding env Variable PYTHONNOUSERSITE to ${getDisplayPath(interpreter?.uri)}`);
-            mergedVars.PYTHONNOUSERSITE = 'True';
-        }
-        if (isPythonKernel) {
-            mergedVars.PYDEVD_IPYTHON_COMPATIBLE_DEBUGGING = '1';
+        if (isPythonKernel || interpreter) {
+            // Merge the env variables with that of the kernel env.
+            interpreterEnv = interpreterEnv || {};
+
+            if (this.configService.getSettings(resource).useOldKernelResolve) {
```

Comment:
Simple fallback in case users run into issues, will be removed in a later release (debt item)